src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/DRAGENGenotypesModel.java

```diff
@@ -0,0 +1,325 @@
+                        FLAT_SNP_HET_PRIOR, api, maxEffectiveDepthAdjustment, calculators);
+                if (genotyperDebugStream != null) {
+                    genotyperDebugStream.println("FRD results:");
+                    genotyperDebugStream.println(Arrays.toString(FRDCallResults));
+                }
+            }
+
+            //make synthesized likelihoods object (NOTE that we can do this since for invalid model GT fields we simply infinity out the result in the array)
+            for (int gt = 0; gt < ployidyModelGenotypeLikelihoods.length; gt++) {
+                if (computeBQD) {
+                    ployidyModelGenotypeLikelihoods[gt] = Math.max(ployidyModelGenotypeLikelihoods[gt], BQDCallResults[gt]);
+                }
+                if (computeFRD) {
+                    ployidyModelGenotypeLikelihoods[gt] = Math.max(ployidyModelGenotypeLikelihoods[gt], FRDCallResults[gt]);
+                }
+            }
+
+            // this is what the work actually is, after we have computed a few things
+            genotypeLikelihoods.add(GenotypeLikelihoods.fromLog10Likelihoods(ployidyModelGenotypeLikelihoods));
+            if (genotyperDebugStream != null) {
+                genotyperDebugStream.println("merged matrix:");
+                genotyperDebugStream.println(Arrays.toString(ployidyModelGenotypeLikelihoods));
+            }
+        }
+        return new GenotypingLikelihoods<>(genotypingAlleles, ploidyModel, genotypeLikelihoods);
+    }
+
+    // Adding the debug stream managed by the haplotype caller engine
+    @Override
+    public void addDebugOutStream(final PrintStream debugStream) {
+        this.genotyperDebugStream = debugStream;
+    }
+
+    private GenotypeLikelihoodCalculatorDRAGEN getLikelihoodsCalculator(final int samplePloidy, final int alleleCount) {
+        if (samplePloidy >= cachePloidyCapacity || alleleCount >= cacheAlleleCountCapacity) {
+            return calculators.getInstanceDRAGEN(samplePloidy, alleleCount);
+        }
+        final GenotypeLikelihoodCalculatorDRAGEN result = likelihoodCalculators[samplePloidy][alleleCount];
+        if (result != null) {
+            return result;
+        } else {
+            final GenotypeLikelihoodCalculatorDRAGEN newOne = calculators.getInstanceDRAGEN(samplePloidy, alleleCount);
+            likelihoodCalculators[samplePloidy][alleleCount] = newOne;
+            return newOne;
+        }
+    }
+
+    /**
+     * This helper class is used to store the necessary data in order to sort a read based on its BQD "feather end" as
+     * well as information relevant to re-associate the read with its position in the AlleleLikelihoods object arrays.
+     */
+    static class DragenReadContainer {
+        final public GATKRead underlyingRead;
+        final int offsetIntoReadForBaseQuality;
+        final int unclippedEnd;
+        final int indexInLikelihoodsObject;
```

Comment:
No there is currently no safeguard... You are right that this could be a problem. This container only lives for the duration of a genotype call though perhaps there is a way to enforce the matching.